src/db.c

```diff
@@ -2004,6 +2027,31 @@ db_files_get_count_bymatch(char *path)
 #undef Q_TMPL
 }
 
+int
+db_file_get_seekpos(uint32_t id)
```

Comment:
I guess this is the trade-off between code reuse and optimization. I would go for reuse unless there is an actual performance bottleneck.

There are probably other places the entire mfi is fetched despite only a single int being looked for (like data_kind). I don't want db_file_get_x, _y and _z. Do you think it is possible to make a generic version, something like db_file_get_int(fieldref)? Then the fieldref probably shouldn't be the actual column name, but perhaps using the offset of mfi->field could work as an argument?